src/codegen/c.zig

```diff
@@ -1399,9 +1400,28 @@ fn airAlloc(f: *Function, inst: Air.Inst.Index) !CValue {
     const writer = f.object.writer();
     const inst_ty = f.air.typeOfIndex(inst);
 
-    // First line: the variable used as data storage.
     const elem_type = inst_ty.elemType();
     const mutability: Mutability = if (inst_ty.isConstPtr()) .Const else .Mut;
+    if (!elem_type.hasCodeGenBits()) {
+        const target = f.object.dg.module.getTarget();
+        const local_value = f.allocLocalValue();
+        const const_prefix = switch (mutability) {
+            .Const => "const ",
+            .Mut => "",
+        };
+        try writer.print("void * {s}", .{const_prefix});
+        try f.object.dg.writeCValue(writer, local_value);
+        try writer.writeAll(" = (void *)");
+        switch (target.cpu.arch.ptrBitWidth()) {
+            32 => try writer.writeAll("0xaaaaaaaa"),
+            64 => try writer.writeAll("0xaaaaaaaaaaaaaaaa"),
+            else => unreachable,
+        }
+        try writer.writeAll(";\n");
+        return CValue{ .local_ref = local_value.local };
+    }
```

Comment:
There should be no local variable allocated in this case. See the `airAlloc` function in the LLVM backend for inspiration.